src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -22,17 +23,43 @@ namespace Microsoft.CodeAnalysis.CSharp.QuickInfo
     [ExtensionOrder(After = QuickInfoProviderNames.Semantic)]
     internal class CSharpSyntacticQuickInfoProvider : CommonQuickInfoProvider
     {
+        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
         [ImportingConstructor]
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-        public CSharpSyntacticQuickInfoProvider()
+        public CSharpSyntacticQuickInfoProvider(IDiagnosticAnalyzerService diagnosticAnalyzerService)
         {
+            _diagnosticAnalyzerService = diagnosticAnalyzerService;
```

Comment:
this is no longer syntactic.  i would make into another provider.

Reply:
Is it okay to rely on this service here? The idea to use this service was taken from #44848 and it seemed to me that this service adds a strong dependency to the IDE.